float32 degree
---
bool success